lib/screens/speaker/speaker_repository.dart

```diff
@@ -0,0 +1,46 @@
+import 'dart:math';
+
+import 'package:ecellapp/core/res/errors.dart';
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+
+@immutable
+abstract class SpeakerRepository {
+  /// All subfunctions are final No arguments required returns json
+  Future<List<dynamic>> speakers();
```

Comment:
use the `fromJson()` function created in the `Speaker` class and return list of speakers
```suggestion
  Future<List<Speaker>> speakers();
```